migrations/20231212184028_indexes.sql

```diff
@@ -0,0 +1,9 @@
+-- Redundant with finished_at
+ALTER TABLE
+    job_queue DROP COLUMN completed_at;
+
+-- Needed for `dequeue` queries
+CREATE INDEX idx_job_queue_status_scheduled_at_queue ON job_queue(status, scheduled_at, queue);
+
+-- Needed for UPDATE-ing incomplete jobs with a specific target (i.e. slow destinations)
+CREATE INDEX idx_job_queue_status_target_queue ON job_queue(status, target, queue);
```

Comment:
I think I would rather have `queue` as the left-most column: Consumers (as of right now) can only listen on one queue, so will always filter on `queue = $queue_name`. Given that constraint will always be an equality constraint, we should keep it in the leading columns to ensure PG uses it (whereas `scheduled_at` for example will likely be a inequality)

Reply:
Interesting, I did some searches o refresh myself when I wrote this, and the advice seemed to be "put the most restrictive filter first", and I figured the table would typically be full of `completed` events, or retry events that are in the future, and so `(status, scheduled_at)` would tend to get to a small number of rows that `queue` would then need to filter.

Of course I didn't do any testing or anything, and I'm not sure it's worth taking that detour right now since we can also just change the index around and see how it goes pretty easily.

Either way, I'm totally fine changing it, just wanted to write down what I was thinking.